tests/unit/test_utils.py

```diff
@@ -120,3 +122,67 @@ def test_returns_none_if_event_is_too_short(self):
         service, operation = find_service_and_method_in_event_name(event_name)
         self.assertIs(service, None)
         self.assertIs(operation, None)
+
+
+class TestOutputStreamFactory(unittest.TestCase):
+    def setUp(self):
+        self.popen = mock.Mock(subprocess.Popen)
+        self.stream_factory = OutputStreamFactory(self.popen)
+
+    @mock.patch(
+        'awscli.utils.get_popen_kwargs_for_pager_cmd')
+    def test_pager(self, mock_get_popen_pager):
+        mock_get_popen_pager.return_value = {
+                'args': ['mypager', '--option']
+        }
+        with mock.patch('os.environ', {}):
```

Comment:
I think we can get rid of this patch now that the environment has been hoisted and maybe move a test that relies on the environment variable to both `test_show.py` and `test_list.py`?